src/components/Sections/GoogleMapsSection/index.tsx

```diff
@@ -0,0 +1,199 @@
+    },
+    {
+      featureType: "poi.park",
+      elementType: "geometry",
+      stylers: [{ color: "#dedede" }, { lightness: 21 }],
+    },
+    {
+      elementType: "labels.text.stroke",
+      stylers: [{ visibility: "on" }, { color: "#ffffff" }, { lightness: 16 }],
+    },
+    {
+      elementType: "labels.text.fill",
+      stylers: [{ saturation: 36 }, { color: "#333333" }, { lightness: 40 }],
+    },
+    { elementType: "labels.icon", stylers: [{ visibility: "off" }] },
+    {
+      featureType: "transit",
+      elementType: "geometry",
+      stylers: [{ color: "#f2f2f2" }, { lightness: 19 }],
+    },
+    {
+      featureType: "administrative",
+      elementType: "geometry.fill",
+      stylers: [{ color: "#fefefe" }, { lightness: 20 }],
+    },
+    {
+      featureType: "administrative",
+      elementType: "geometry.stroke",
+      stylers: [{ color: "#fefefe" }, { lightness: 17 }, { weight: 1.2 }],
+    },
+  ] as google.maps.MapTypeStyle[];
+  map: google.maps.Map | null = null;
+
+  @Prop() title: GoogleMapsSectionProps["title"];
+  @Prop({ required: false }) apikey!: GoogleMapsSectionProps["apikey"];
+  @Prop({ required: false })
+  startLocation!: GoogleMapsSectionProps["startLocation"];
+  @Prop({ required: false, default: 15 }) zoom!: GoogleMapsSectionProps["zoom"];
+  @Prop({ required: false, default: "en" })
+  language!: GoogleMapsSectionProps["language"];
+  @Prop({ required: false }) locations!: GoogleMapsSectionProps["locations"];
+  @Prop({ required: false, default: "Contact Us" })
+  buttonLabel!: GoogleMapsSectionProps["buttonLabel"];
+  @Prop({ required: true })
+  buttonHandle!: GoogleMapsSectionProps["buttonHandle"];
+
+  placeMarkers(
+    map: google.maps.Map,
+    ...locations: MapsLocation[]
+  ): google.maps.Marker[] {
+    const renderDescriptionBox = (location: MapsLocation): Node => {
+      const template = `<h2 class="font-bold">${location.name}</h2>
+        <p>${location.street}</p>
+        ${location.description ? "<p>" + location.description + "</p>" : ""}
+        <p>${location.city}</p>
+        <button class="google-maps-description-button">
+        ${this.buttonLabel}
+        </button>`;
+      const div = document.createElement("div");
+      div.classList.add("google-maps-description");
+      div.innerHTML = template;
+      div.querySelector("button")?.addEventListener("click", event => {
+        this.buttonHandle(event, location);
+      });
+      return div;
+    };
+
+    return locations.map((location: MapsLocation) => {
+      const marker = new google.maps.Marker({
+        position: location.position,
+        icon: markerIcon,
+        map,
+      });
+      const infoWindow = new google.maps.InfoWindow({
+        content: renderDescriptionBox(location),
+      });
+
+      marker.addListener("click", () => {
+        infoWindow?.open(map, marker);
+      });
+      return marker;
+    });
+  }
+
+  getStartingLocation(callback: Function): void {
```

Comment:
When these 3 methods are combined into one you do not need to add map as local state.

Reply:
And it can be easier to follow the flow of the initialization.